server/coordinator/shard_picker.go

```diff
@@ -99,3 +100,63 @@ func copyNodeShardMapping(nodeShardsMapping map[string][]storage.ShardNode) map[
 	}
 	return tempNodeShardMapping
 }
+
+// LeastTableShardPicker selects shards based on the number of tables on the current shard,
+// and always selects the shard with the smallest number of current tables.
+type LeastTableShardPicker struct{}
+
+func NewLeastTableShardPicker() ShardPicker {
+	return &LeastTableShardPicker{}
+}
+
+func (l LeastTableShardPicker) PickShards(_ context.Context, snapshot metadata.Snapshot, expectShardNum int) ([]storage.ShardNode, error) {
+	shardNodeMapping := make(map[storage.ShardID]storage.ShardNode, len(snapshot.Topology.ShardViewsMapping))
+	for _, shardNode := range snapshot.Topology.ClusterView.ShardNodes {
+		shardNodeMapping[shardNode.ID] = shardNode
+	}
+
+	if len(snapshot.Topology.ClusterView.ShardNodes) == 0 {
+		return nil, errors.WithMessage(ErrNodeNumberNotEnough, "no shard is assigned")
+	}
+
+	result := make([]storage.ShardNode, 0, expectShardNum)
+
+	shardTableNumberMapping := make(map[storage.ShardID]int, len(snapshot.Topology.ShardViewsMapping))
+	for shardID, shardView := range snapshot.Topology.ShardViewsMapping {
+		shardTableNumberMapping[shardID] = len(shardView.TableIDs)
+	}
+
+	var sortedMapping []Pair
+	for i := 0; i < expectShardNum; i++ {
+		if i%len(shardTableNumberMapping) == 0 {
+			sortedMapping = sortByTableNumber(shardTableNumberMapping)
+		}
+
+		selectShard := sortedMapping[i%len(shardTableNumberMapping)]
+		shardTableNumberMapping[selectShard.shardID]++
+
+		result = append(result, shardNodeMapping[selectShard.shardID])
+	}
```

Comment:
I guess it can be made in an easier and concise way: Pick the `expectShardNum` shards with the fewest tables, that is to say, there is no need to maintain the shards' table number after one shard selected.

And with this way, the selected `expectShardNum` is also ensured different.